CMakeLists.txt

```diff
@@ -10,17 +10,18 @@ if(NOT CMAKE_BUILD_TYPE)
         FORCE)
 endif(NOT CMAKE_BUILD_TYPE)
 
-set(HDF5_ECF_PLUGIN_BUILD_PATH "${CMAKE_BINARY_DIR}/lib/hdf5/plugin" CACHE INTERNAL "" FORCE)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib/hdf5/plugin)
+
 if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT AND NOT CMAKE_INSTALL_PREFIX)
     if(APPLE)
-        set(HDF5_ECF_PLUGIN_INSTALL_PATH "/opt/local/lib/hdf5/" CACHE INTERNAL "" FORCE)
+        set(HDF5_ECF_PLUGIN_INSTALL_PATH "local/lib/hdf5/" CACHE INTERNAL "" FORCE)
     elseif(WIN32)
-        set(HDF5_ECF_PLUGIN_INSTALL_PATH "$ENV{ALLUSERSPROFILE}/hdf5/lib/plugin/" CACHE INTERNAL "" FORCE)
+        set(HDF5_ECF_PLUGIN_INSTALL_PATH "hdf5/lib/plugin/" CACHE INTERNAL "" FORCE)
     else()
-        set(HDF5_ECF_PLUGIN_INSTALL_PATH "/usr/local/hdf5/lib/plugin" CACHE INTERNAL "" FORCE)
+        set(HDF5_ECF_PLUGIN_INSTALL_PATH "local/hdf5/lib/plugin" CACHE INTERNAL "" FORCE)
```

Comment:
here too, I'm having trouble understanding the changes, if the user did not specify a `CMAKE_INSTALL_PREFIX`, we force the installation path to be the correct default path for HDF5 plugins (otherwise the plugin can not be found).
On the other hand, if `CMAKE_INSTALL_PREFIX` is set, the l.24 is redudant but it should do the correct thing which is to install in the provided prefix path, no ?

Reply:
For me the issue with working with `CMAKE_INSTALL_PREFIX` is that it duplicates the usage of this variable. The whole idea of `CMAKE_INSTALL_PREFIX` is that it will be prepended to all install paths (https://cmake.org/cmake/help/latest/variable/CMAKE_INSTALL_PREFIX.html). Then the only reason I see of changing 
it in CMake files is when default path must be avoided. But the default for linux is `/usr/local` so it duplicates the usage.

Basically I think we always end up with 

```
install(TARGETS hdf5_ecf_plugin
        RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}/something" COMPONENT hdf5-plugin-ecf-local-install
        LIBRARY DESTINATION "${CMAKE_INSTALL_PREFIX}/something" COMPONENT hdf5-plugin-ecf-local-install
``` 

And I think we should avoid having any `CMAKE_INSTALL_PREFIX` ending up in `install`. 

Also I don't know if it's expected but when looking at the CMakeLists.txt I just saw differences in unix paths:
`/usr/local/hdf5/lib/plugin` (`hdf5/lib/plugin`) or `${CMAKE_INSTALL_PREFIX}/lib/hdf5/plugin`  (`lib/hdf5/plugin`).

Do you know if the Metavision is using this plugin so I can check if it broke something?

Reply:
The first part of the `if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT AND NOT CMAKE_INSTALL_PREFIX)` should stay as it is (it forces the installation path if the user did not specify one). The `else` part can be changed as you are suggesting, I agree that there is no point in prepending the prefix.
The plugin is definitely used in MV as this repo is a submodule and it ends up being compiled as part of the SDK driver module.